Kernel/Syscalls/mmap.cpp

```diff
@@ -69,8 +69,12 @@ static bool should_make_executable_exception_for_dynamic_loader(bool make_readab
     return true;
 }
 
-static bool validate_mmap_prot(int prot, bool map_stack, bool map_anonymous, Memory::Region const* region = nullptr)
+static bool validate_mmap_prot(const Process& process, int prot, bool map_stack, bool map_anonymous, Memory::Region const* region = nullptr)
 {
+    bool wxallowed = process.executable()->mount_flags() & MS_WXALLOWED;
+    if (wxallowed)
+        return true;
```

Comment:
This means we allow other forbidden protection combinations as well, like executable&stack or executable&anonymous.  We most definitely don't want that :^). The check should probably go before line 81 (85) to skip the W^X check if we have wxallowed.

Reply:
Good catch, I've moved so that it just skips w^x checks